pkg/k8s/metadata.go

```diff
@@ -0,0 +1,102 @@
+package k8s
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"encoding/json"
+	"github.com/feedhenry/mcp-standalone/pkg/mobile"
+	"github.com/pkg/errors"
+)
+
+type Metadata struct {
+	Issuer                        string   `json:"issuer"`
+	AuthorizationEndpoint         string   `json:"authorization_endpoint"`
+	TokenEndpoint                 string   `json:"token_endpoint"`
+	ScopesSupported               []string `json:"scopes_supported"`
+	ResponseTypesSupported        []string `json:"response_types_supported"`
+	GrantTypesSupported           []string `json:"grant_types_supported"`
+	CodeChallengeMethodsSupported []string `json:"code_challenge_methods_supported"`
+}
+
+func GetMetadata(k8shost string, requester mobile.ExternalHTTPRequester) (*Metadata, error) {
+	url := fmt.Sprintf("%s%s", k8shost, "/.well-known/oauth-authorization-server")
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create request to retrieve OpenShift server metadata")
+	}
+	res, err := requester.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to do request to retrieve OpenShift server metadata")
+	}
+	defer res.Body.Close()
+	if res.StatusCode != 200 {
+		data, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return nil, errors.Wrap(err, "attempted to read body after failed OpenShift server metadata call")
+		}
+		if nil != data {
+			return nil, errors.Wrap(err, "unexpected response from OpenShift")
```

Comment:
maybe log out the data or add it to the error message ``` " + string(data) ```

Reply:
sure :+1: